### ISSUE TITLE
[constructor-super] Duplicates TypeScript's TS2335 and TS2377 errors

### ISSUE THREAD

**Issue by glen-84** [typescript-eslint/typescript-eslint#1248]
**Repro**

```JSON
{
  "rules": {
    "constructor-super": "error"
  }
}
```

```TS
class A {
    public constructor() {
        super();
    }
}

class B extends A {
    public constructor() {
        // ...
    }
}
```

**Expected Result**

TS errors only.

**Actual Result**

TS and ESLint errors.

![image](https://user-images.githubusercontent.com/261509/69424303-f6e4c300-0d30-11ea-827a-0fb323bc6d8b.png)

**Additional Info**

This relates to the recommended config (`configs/eslint-recommended.ts`).

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.8.0` |
| `@typescript-eslint/parser`        | `2.8.0` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12.13.0` |
| `npm`                              | `6.12.0` |

**Comment by bradzacher** [comment]
consolidated eslint-recommended changes in #1273
